maven-extension/src/main/java/io/opentelemetry/maven/OpenTelemetrySdkService.java

```diff
@@ -47,26 +56,69 @@ public OpenTelemetrySdkService() {
             .setServiceClassLoader(getClass().getClassLoader())
-            .addPropertiesSupplier(() -> properties)
+            .addPropertiesCustomizer(
+                configProperties -> {
+                  // The OTel SDK by default sends data to the OTLP gRPC endpoint at localhost:4317.
+                  // Change this behavior to disable by default the OTel SDK in the Maven extension
+                  // so
+                  // that it must be explicitly enabled by the user.
+                  // To change this default behavior, we set "otel.[traces,metrics,logs].exporter"
+                  // to
+                  // "none" if the endpoint has not been specified
+                  if (configProperties.getString("otel.exporter.otlp.endpoint") == null) {
+                    Map<String, String> properties = new HashMap<>();
+                    if (Objects.equals(
+                            "otlp", configProperties.getString("otel.traces.exporter", "otlp"))
+                        && configProperties.getString("otel.exporter.otlp.traces.endpoint")
+                            == null) {
+                      properties.put("otel.traces.exporter", "none");
+                    }
+                    if (Objects.equals(
+                            "otlp", configProperties.getString("otel.metrics.exporter", "otlp"))
+                        && configProperties.getString("otel.exporter.otlp.metrics.endpoint")
+                            == null) {
+                      properties.put("otel.metrics.exporter", "none");
+                    }
+                    if (Objects.equals(
+                            "otlp", configProperties.getString("otel.logs.exporter", "otlp"))
+                        && configProperties.getString("otel.exporter.otlp.logs.endpoint") == null) {
+                      properties.put("otel.logs.exporter", "none");
+                    }
+                    return properties;
+                  } else {
+                    return Collections.emptyMap();
+                  }
+                })
+            .addPropertiesCustomizer(
+                config -> {
+                  // keep a reference to the computed config properties for future use in the
+                  // extension
+                  this.configProperties = config;
+                  return Collections.emptyMap();
+                })
+            .addResourceCustomizer(
+                (res, configProperties) -> {
+                  // keep a reference to the computed Resource for future use in the extension
+                  this.resource = Resource.builder().putAll(res).build();
+                  return this.resource;
+                })
             .disableShutdownHook()
             .build();
 
+    if (this.resource == null) {
+      this.resource = Resource.empty();
+    }
+    if (this.configProperties == null) {
+      this.configProperties = DefaultConfigProperties.createFromMap(Collections.emptyMap());
+    }
```

Comment:
[minor] in order to avoid any ambiguity and possible concurrent modification by the SDK init (I'm pretty sure it's synchronous but we'd better be safe than sorry), it would be better to set `this.resource` and `this.configProperties` before the call to `AutoConfiguredOpenTelemetrySdk.builder()`.